### PR TITLE
feature/assignment14

### PR DESCRIPTION
**Pull request by dheerajsrinath**
Create a class Actor which has one method work() in that method write System.out.println(“Doing
acting”), After that Create class Director which has one method work() in that method write
System.out.println(“Directing”) and extends Actor class in Director class like = class Director
extends Actor.
Call both the method and print in console = Doing acting, Directing
Note - Method overriding concept

### CODE REVIEW

File: src/main/java/Assignment14.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Actor{
+    public void work(){
+        System.out.println("Doing Acting");
+    }
+}
+class Director {
```
Comment (PranavHanwante) [inline src/main/java/Assignment14.java]: Intent

File: src/main/java/Assignment14.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    public void work(){
+        System.out.println("Directing");
+    }
+}
+public class Assignment14 {
```
Comment (PranavHanwante) [inline src/main/java/Assignment14.java]: Intent

File: src/main/java/Assignment14.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    public static void main(String[]args){
+        Actor act = new Actor();
+        Director dir = new Director();
+        act.work();
+        dir.work();
```
Comment (PranavHanwante) [inline src/main/java/Assignment14.java]: you can direclty write like this too
new Actor().work();
new Director().work();

File: src/main/java/com/assignment/Actor.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.assignment;
+
+public class Actor {
+    public String work() {
```
Comment (PranavHanwante) [inline src/main/java/com/assignment/Actor.java:4]: intend

Comment (PranavHanwante) [inline src/main/java/com/assignment/Actor.java]: intent

File: src/main/java/com/assignment/Director.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.assignment;
+
+public class Director extends Actor{
+    public String work() {
```
Comment (PranavHanwante) [inline src/main/java/com/assignment/Director.java]: intent

File: src/test/java/com/assignmentTest/Assignment14Test.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Assignment14Test {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Test
+    public void methodOverloading() {
```
Comment (PranavHanwante) [inline src/test/java/com/assignmentTest/Assignment14Test.java]: test word should be present in methodOverloading, and method overloading is an oops concept so change the method name

File: src/test/java/com/assignmentTest/Assignment14Test.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        Actor actor = new Actor();
+        Director director = new Director();
+
+        logger.info(actor.work());
+        Assertions.assertEquals(actor.work(),"Doing Acting","Action doesn't Match");
+
+        logger.info(director.work());
+        Assertions.assertEquals(director.work(),"Directing","Action doesn't Match");
```
Comment (PranavHanwante) [inline src/test/java/com/assignmentTest/Assignment14Test.java]: "Action doesn't Match message" is used two times so put that in string and use the string here

**Review by PranavHanwante — APPROVED** [review]

File: src/main/java/com/assignment/Director.java [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package com.assignment;
+
+public class Director extends Actor {
+
+    public String work() {
```
Comment (akhil7000) [inline src/main/java/com/assignment/Director.java]: violates the method naming

File: src/main/java/com/assignment/Actor.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package com.assignment;
+
+public class Actor {
+
+    public String work() {
```
Comment (akhil7000) [inline src/main/java/com/assignment/Actor.java]: violates method naming

File: src/test/java/com/training/sample/tests/SampleTest.java [hunk 10]
```diff
@@ -1,22 +0,0 @@
-package com.training.sample.tests;
```
Comment (akhil7000) [inline src/test/java/com/training/sample/tests/SampleTest.java]: Why is this file deleted?
Please do not delete any sample files/code

**Review by akhil7000 — CHANGES_REQUESTED** [review]

**Review by akhil7000 — APPROVED** [review]

**Pull request merged**